### DISCUSSION TITLE
getRecords returns string instead of array of Records

### DISCUSSION THREAD

**Discussion by DevHeaven** [spatie/dns#89] [Q&A]
Hi Spatie

I might be overlooking the obvious here, but as I read in the docs, getRecords() should return an array of Records
however, when testing it returns a string with all TXT records.

**code used**
$records = Dns::of('example.com','8.8.8.8')->getRecords('TXT');

**Expected result**
$records = ["yxvy9m4blrswgrsz8ndjh467n2y7mgl2","v=spf1 -all"] 

**Result**
"""
example.com.\t\t17794 IN TXT "v=spf1 -all"\r\n
example.com.\t\t17794 IN TXT "yxvy9m4blrswgrsz8ndjh467n2y7mgl2"\r\n
"""

**Comment by ryantxr** [comment]
You are probably using the same version as I am. 1.6, which returns everything in a string and you have to parse it.
In my case, I only need the record type so I use this code snippet. It is a little crude but I have used it for several hundred domains.
```
try {
            $dns = DnsService::of($domain, '1.1.1.1');
            $rr = $dns->getRecords(); // returns all available dns records
            $all = explode("\n", $rr);
            $len = strlen($domain);
        
            foreach ( $all as $a ) {
                $t1 = explode("\t", $a, 2);
                if ( substr($t1[0], 0, $len) != $domain ) {
                    continue;
                }
                
                if ( empty($a) ) {
                    continue;
                }
                // "dmesolutions.com.\t10800 IN CNAME traff-1.hugedomains.com."
                // "gmail.com.\t\t219 IN A 142.250.189.133"
                if ( preg_match('/(.+)[ \t]{1,2}([0-9]+) IN[ \t]([A-Z]+)/', $a, $matches) ) {
                    if ( in_array($matches[1], ['A', 'CNAME'] ) ) {
                        // found CNAME or A record
                        return true;
                    } else {
                        continue;
                    }
                } else {
                    echo "\nbad preg\n";
                    return false;
                }
            }
        } catch ( \Spatie\Dns\Exceptions\CouldNotFetchDns $e ) {
            return false;
        }
```

**Comment by Gummibeer** [comment]
With `spatie/dns:2*` this whole thing changed - the library was fully re-written. So I assume that this discussion is based on mismatch of used version and docs.
